javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -190,7 +186,7 @@ private Vector(BitMappedTrie<T> trie) {
      * @param array a boolean array
      * @return A new Vector of Boolean values
      */
-    public static Vector<Boolean> ofAll(boolean[] array) {
+    public static Vector<Boolean> ofAll(boolean... array) {
```

Comment:
I tested it with Stream (short version):

```java
interface Stream<T> {

    static Stream<Integer> ofAll(int... array) {
        return Stream.ofAll(Iterator.ofAll(array));
    }

    static Stream<Long> ofAll(long... array) {
        return Stream.ofAll(Iterator.ofAll(array));
    }
}
```

The test:

```java
public class Test {{

    // ok
    Stream<Integer> ints = Stream.ofAll();

    // IntelliJ IDEA complains: Required: Stream<Long>, Found: Stream<Integer>
    Stream<Long> longs = Stream.ofAll();

    // does also not work
    Stream<Long> longs = Stream.<Long> ofAll();
}}
```

However, unexpected behavior with varargs can only occur when we have reference types, e.g.

```java
// reference type T
public <T> void foo(T... params) { ... }

int[] arr = {1, 2, 3};

foo(arr); // passes an int[][] array containing a single int[] element
```

See also [this StackOverflow question](http://stackoverflow.com/questions/5405673/java-varags-method-param-list-vs-array).

I see no problem to switch to varargs for primitive types. There is no risk that the array is confused with a single object.

We should apply that change then to all other Javaslang types that have static factory methods `ofAll(<primitive array>)` - of course in another PR.

That's a nice API change, simplifying things! :-)